azure/src/test/java/com/microsoft/azure/AzureTests.java

```diff
@@ -61,30 +63,44 @@ public void setup() throws Exception {
         // Authenticate based on file
         this.azure2 = Azure.authenticate(new File("my.azureauth"))
                 .withDefaultSubscription();
+        
     }
 
     /**
      * Tests the public IP address implementation
      * @throws Exception
      */
     @Test public void testPublicIpAddresses() throws Exception {
-        new TestPublicIpAddress().runTest(azure2.publicIpAddresses());        
+        new TestPublicIpAddress().runTest(azure2.publicIpAddresses(), azure2);        
     }
 
     /**
+     * Tests the public IP address implementation from an individual service client
+     * @throws IOException 
+     * @throws CloudException 
+     */
+    @Test
+    public void testPublicIpAddressesFromServiceClient() throws IOException, CloudException {
+        NetworkManager networkManager= NetworkManager.authenticate(
+                ApplicationTokenCredentials.fromFile(new File("my.azureauth")), 
+                "9657ab5d-4a4a-4fd2-ae7a-4cd9fbd030ef");
+        
+    }
+    
```

Comment:
that'd just because I had already started working on the next checkin and just didn't remove it before updating this PR, since it doesn't hurt. But my next checkin just added to the PR has more of that.